client.go

```diff
@@ -419,7 +423,18 @@ func unixDial(networ, addr string) (net.Conn, error) {
 			return nil, fmt.Errorf(gettext.Gettext("cannot resolve unix socket address: %v"), err)
 		}
 	}
-	return net.DialUnix("unix", nil, raddr)
+	result, err = net.DialUnix("unix", nil, raddr)
+
+	switch t := err.(type) {
+	case *net.OpError:
```

Comment:
I don't think this is quite right. You can also get a permissions error or a connection refused error, but in both of those cases LXD might be running.